src/GT.Data/Repositories/Impl/GTGenericRepository.cs

```diff
@@ -1,50 +1,28 @@
 ﻿using GT.Data.Data;
+using GT.Data.Data.GTAppDb;
+using GT.Data.Repositories.Interfaces;
 using Microsoft.EntityFrameworkCore;
-using Microsoft.EntityFrameworkCore.Query;
-using System.Linq.Expressions;
 
-namespace GT.Data.Repositories
+namespace GT.Data.Repositories.Impl
 {
 	/// <summary>
 	/// Base repository class.
 	/// </summary>
 	/// <typeparam name="TEntity">The database entity handled by the repository.</typeparam>
 	public class GTGenericRepository<TEntity>
-	: IGTGenericRepository<TEntity>, IDisposable
```

Comment:
In hindsight it could be worth reconsidering getting an implementation of IDisposable to work to simplify service constructors/structure by allowing them to access extra repositories when needed, for example letting ListingService access the Company Repository while creating a new listing.

Reply:
Good idea, I'll create an issue for that.